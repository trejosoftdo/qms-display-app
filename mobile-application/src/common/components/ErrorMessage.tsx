import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { StyleSheet } from 'react-native';
import { Text } from 'react-native-paper';
import AppLightTheme from '../theme';


/**
 * ConditionalContainerProps defines the props for the Conditional Container Component.
 */
interface ErrorMessageProps {}

/**
 * A component that display a general error message when an unexpected error occurrs
 *
 * @param {ErrorMessageProps} props - The props for the Error Message component.
 */
const ErrorMessage: React.FC<ErrorMessageProps> = (props: ErrorMessageProps) => {
  const { t } = useTranslation();
  return (
    <Text style={styles.message} variant="titleMedium">
      {t('translation:defaultErrorMessage')}
    </Text>
  );
};

const styles = StyleSheet.create({
  message: {
    color: AppLightTheme.colors.error,
    padding: 16,
    paddingBottom: 32,
    borderWidth: 1,
    borderColor: AppLightTheme.colors.error,
    backgroundColor: AppLightTheme.colors.tertiaryContainer,
    margin: 8,
    borderRadius: 5,
  },
});

export default ErrorMessage;
