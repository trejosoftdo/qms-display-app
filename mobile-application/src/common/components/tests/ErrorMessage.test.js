import React from 'react';
import ErrorMessage from '../ErrorMessage';

jest.mock('react-i18next');

describe('ErrorMessage component', () => {
  it('renders with no provided props as expected', () => {
    expect(renderShallow(<ErrorMessage />)).toMatchSnapshot();
  });
});