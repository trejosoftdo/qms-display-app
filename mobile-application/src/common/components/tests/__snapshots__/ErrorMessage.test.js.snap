// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component renders with no provided props as expected 1`] = `
<ForwardRef(Text)
  style={
    {
      "backgroundColor": "#feeef4",
      "borderColor": "#cc426a",
      "borderRadius": 5,
      "borderWidth": 1,
      "color": "#cc426a",
      "margin": 8,
      "padding": 16,
      "paddingBottom": 32,
    }
  }
  variant="titleMedium"
>
  Translated[translation:defaultErrorMessage]({})
</ForwardRef(Text)>
`;
